db/db_impl/db_impl.cc

```diff
@@ -1881,6 +1881,8 @@ Status DBImpl::GetImpl(const ReadOptions& read_options, const Slice& key,
       return s;
     }
   }
+  std::shared_ptr<PinnedIteratorsManager> pinned_iters_mgr(
+      new PinnedIteratorsManager());
```

Comment:
Besides the good comment by @ajkr , shared_ptr here doesn't appear to be necessary, as it is never shared. A unique_ptr is more straight-forward, and something allocated in the stack would be even better, if possible.